llama_index/indices/multi_modal/base.py

```diff
@@ -169,6 +182,12 @@ def _get_node_with_embedding(
             embedding = id_to_embed_map[node.node_id]
             result = node.copy()
             result.embedding = embedding
+            if is_image and id_to_text_embed_map:
+                text_embedding = id_to_text_embed_map[node.node_id]
+                result.text_embedding = text_embedding
+                result.embedding = (
+                    text_embedding  # TODO: re-factor to make use of both embeddings
```

Comment:
@hatianzhang:

This is where we'd need to refactor in the future. Currently am overriding embedding field with text_embedding if again text field is populated.